nova/virt/vmwareapi/vmops.py

```diff
@@ -1517,18 +1517,24 @@ def _destroy_instance(self, context, instance, destroy_disks=True):
                                            "UnregisterVM", vm_ref)
                 LOG.debug("Unregistered the VM", instance=instance)
 
-                with lockutils.lock(server_group_info.uuid,
-                                lock_file_prefix='nova-vmware-server-group'):
-                    cluster_config = cluster_util.get_cluster_property(
-                        self._session, "configurationEx", self._cluster)
-                    # Check if the unregistered vm is the last vm in its group
-                    for group in cluster_config.group:
-                        if group.name == server_group_info.uuid:
-                            if not hasattr(group, 'vm'):
-                                cluster_util.delete_vm_group(
-                                    self._session, self._cluster,
-                                    group)
-                            break
+                if server_group_info:
+                    with lockutils.lock(server_group_info.uuid,
+                                    lock_file_prefix='vmware-vm-group-policy'):
```

Comment:
I think this should use the same lock as `cluster_util.update_placement()` as it changes the same thing. Or do you see a need for a group-specific lock?

Reply:
Yeah since I removed the lock in vm_util now we should use the same lock name. So I changed it.